### PR TITLE
fix(curriculum): remove extrernal link to github pug

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

-[x] clear the 3rd task #40749 

 but doesn't Close it

Edit: forgot to update this photo

Pug page

### CODE REVIEW

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ To pass those along from our server, you will need to add an object as a second
 
 It should look like: `res.render(process.cwd() + '/views/pug/index', {title: 'Hello', message: 'Please login'});` Now refresh your page and you should see those values rendered in your view in the correct spot as laid out in your index.pug file!
 
-Submit your page when you think you've got it right. If you're running into errors, you can check out the [project completed](https://gist.github.com/camperbot/4af125119ed36e6e6a8bb920db0c0871).
+Submit your page when you think you've got it right. If you're running into errors, you can check out the project completed to this point [here](https://gist.github.com/camperbot/4af125119ed36e6e6a8bb920db0c0871).
```
Comment (ieahleen) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md:45]: also `here` is not an accessible anchor text, as you are editing this line, please make the anchor text descriptive, maybe even the whole "project completed up to this point" can be the anchor text

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md (hunk 1 above)
Comment (ieahleen) [reply curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md:45]:
```suggestion
Submit your page when you think you've got it right. If you're running into errors, you can check out the [project completed up to this point]((https://gist.github.com/camperbot/4af125119ed36e6e6a8bb920db0c0871)) here.
```

Comment (Sboonny) [reply curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md:45]: There are similar files with `here` as an `anchor`.

 Is it okay to change consistency, or should I change them as well?

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md [hunk 2]
```diff
@@ -12,15 +12,37 @@ One of the greatest features of using a template engine is being able to pass va
 
 In your Pug file, you're able to use a variable by referencing the variable name as `#{variable_name}` inline with other text on an element or by using an equal sign on the element without a space such as `p=variable_name` which assigns the variable's value to the p element's text.
 
-We strongly recommend looking at the syntax and structure of Pug [here](https://github.com/pugjs/pug) on GitHub's README. Pug is all about using whitespace and tabs to show nested elements and cutting down on the amount of code needed to make a beautiful site.
+ Pug is all about using whitespace and tabs to show nested elements and cutting down on the amount of code needed to make a beautiful site. Here is an Example:
+
```
Comment (lasjorg) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md:16]: We might still suggest the camper read the Pug documentation without linking to it?

```suggestion
 Pug is all about using whitespace and tabs to show nested elements and cutting down on the amount of code needed to make a beautiful site. Read the Pug documentation for more information on usage and syntax.
 
 Here is an example: 
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md [hunk 3]
```diff
@@ -12,15 +12,37 @@ One of the greatest features of using a template engine is being able to pass va
...
+  ```
 
 Looking at our pug file 'index.pug' included in your project, we used the variables *title* and *message*.
 
 To pass those along from our server, you will need to add an object as a second argument to your *res.render* with the variables and their values. For example, pass this object along setting the variables for your index view: `{title: 'Hello', message: 'Please login'}`
 
 It should look like: `res.render(process.cwd() + '/views/pug/index', {title: 'Hello', message: 'Please login'});` Now refresh your page and you should see those values rendered in your view in the correct spot as laid out in your index.pug file!
 
-Submit your page when you think you've got it right. If you're running into errors, you can check out the project completed up to this point [here](https://gist.github.com/camperbot/4af125119ed36e6e6a8bb920db0c0871).
+Submit your page when you think you've got it right. If you're running into errors, you can check out the [project completed up to this point](https://gist.github.com/camperbot/4af125119ed36e6e6a8bb920db0c0871) here.
```
Comment (lasjorg) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md:45]:
```suggestion
Submit your page when you think you've got it right. If you're running into errors, you can check out the [project completed up to this point](https://gist.github.com/camperbot/4af125119ed36e6e6a8bb920db0c0871).
```

I think we should remove "here"

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md (hunk 2 above)
Comment (Sboonny) [reply curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md:16]: I do think that the example is insufficient, but people will find inconvenient to be asked to check something without a link to it. I think the best action is to hold this issue too untill an article about pug is written

Comment (lasjorg) [reply curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md:16]: It's really just meant as a small reminder to RTFM. Which people surprisingly often do not.

Comment (Sboonny) [reply curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md:16]: You know what, that is funny. I am in

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md [hunk 4]
```diff
@@ -12,15 +12,38 @@ One of the greatest features of using a template engine is being able to pass va
 
 In your Pug file, you're able to use a variable by referencing the variable name as `#{variable_name}` inline with other text on an element or by using an equal sign on the element without a space such as `p=variable_name` which assigns the variable's value to the p element's text.
 
-We strongly recommend looking at the syntax and structure of Pug [here](https://github.com/pugjs/pug) on GitHub's README. Pug is all about using whitespace and tabs to show nested elements and cutting down on the amount of code needed to make a beautiful site.
+ Pug is all about using whitespace and tabs to show nested elements and cutting down on the amount of code needed to make a beautiful site. Read the Pug documentation for more information on usage and syntax.
+ 
+ Here is an example:
```
Comment (lasjorg) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md:17]: That is my bad, I guess I remove the new line when I made the suggestion. There should be a blank line after "Here is an example:" and before the code block.

https://github.com/updownpress/markdown-lint/blob/master/rules/031-blanks-around-fences.md

**Review by lasjorg — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md (hunk 4 above)
Comment (lasjorg) [reply curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md:17]: dub

**Comment by lasjorg** [comment]
I don't know why I keep getting duplicates?

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md [hunk 5]
```diff
@@ -12,15 +12,38 @@ One of the greatest features of using a template engine is being able to pass va
 
 In your Pug file, you're able to use a variable by referencing the variable name as `#{variable_name}` inline with other text on an element or by using an equal sign on the element without a space such as `p=variable_name` which assigns the variable's value to the p element's text.
 
-We strongly recommend looking at the syntax and structure of Pug [here](https://github.com/pugjs/pug) on GitHub's README. Pug is all about using whitespace and tabs to show nested elements and cutting down on the amount of code needed to make a beautiful site.
+ Pug is all about using whitespace and tabs to show nested elements and cutting down on the amount of code needed to make a beautiful site. Read the Pug documentation for more information on usage and syntax.
+ 
+ Here is an example: 
+ ```html
```
Comment (lasjorg) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md:18]:
```suggestion
 Here is an example: 
 
 ```html
```

That is my bad, I guess I remove the new line when I made the suggestion. There should be a blank line after "Here is an example:" and before the code block.

https://github.com/updownpress/markdown-lint/blob/master/rules/031-blanks-around-fences.md

**Review by lasjorg — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md [hunk 6]
```diff
@@ -12,15 +12,39 @@ One of the greatest features of using a template engine is being able to pass va
...
+  <body>
+    <p>You are amazing</p>
+  </body>
+  ```
 
 Looking at our pug file 'index.pug' included in your project, we used the variables *title* and *message*.
 
 To pass those along from our server, you will need to add an object as a second argument to your *res.render* with the variables and their values. For example, pass this object along setting the variables for your index view: `{title: 'Hello', message: 'Please login'}`
 
 It should look like: `res.render(process.cwd() + '/views/pug/index', {title: 'Hello', message: 'Please login'});` Now refresh your page and you should see those values rendered in your view in the correct spot as laid out in your index.pug file!
```
Comment (lasjorg) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md:45]:
```suggestion
Looking at our pug file `index.pug` included in your project, we used the variables `title` and `message`.

To pass those along from our server, you will need to add an object as a second argument to your `res.render` with the variables and their values. For example, pass this object along setting the variables for your index view: `{title: 'Hello', message: 'Please login'}`

It should look like: `res.render(process.cwd() + '/views/pug/index', {title: 'Hello', message: 'Please login'});` Now refresh your page and you should see those values rendered in your view in the correct spot as laid out in your `index.pug` file!
```
I think the variables `title` and `message`, and the `res.render` method should be in code blocks and not italicized. The same might be said for the `index.pug` file.

If you want to make that change go ahead.

Comment (Sboonny) [reply curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md:45]: Nice catch👍

**Review by lasjorg — APPROVED** [review]
LGTM

Thank you for the extra work. 👍

**Review by raisedadead — APPROVED** [review]

**Pull request merged**